Diagnostics/FreeBusyChecker/Functions/OnPremOAuthFunctions.ps1

```diff
@@ -0,0 +1,648 @@
+        $Script:tdCurrentCertificateStatusColor = "green"
+    } else {
+        Write-Host -ForegroundColor Red "  Certificate is not Valid"
+        $Script:tdCurrentCertificateStatus = "  Certificate is Not Valid"
+        $Script:tdCurrentCertificateStatusColor = "red"
+    }
+    if ($CurrentCertificate.Subject -like "CN=Microsoft Exchange Server Auth Certificate" ) {
+        Write-Host " Subject: " $CurrentCertificate.Subject
+        Write-Host -ForegroundColor Green "  Subject is CN=Microsoft Exchange Server Auth Certificate"
+        $Script:tdCurrentCertificateSubject = "  Subject is CN=Microsoft Exchange Server Auth Certificate"
+        $Script:tdCurrentCertificateSubjectColor = "green"
+    } else {
+        Write-Host -ForegroundColor Red "  Subject is not CN=Microsoft Exchange Server Auth Certificate"
+        $Script:tdCurrentCertificateSubject = "  $($CurrentCertificate.Subject) - Subject should be CN=Microsoft Exchange Server Auth Certificate"
+        $Script:tdCurrentCertificateSubjectColor = "red"
+    }
+    Write-Host -ForegroundColor White "`n Checking Exchange Auth Certificate Distribution `n"
+    $CheckAuthCertDistribution = foreach ($name in (Get-ExchangeServer).name) { Get-ExchangeCertificate -Thumbprint (Get-AuthConfig).CurrentCertificateThumbprint -Server $name -ErrorAction SilentlyContinue | Select-Object Identity, thumbprint, Services, subject }
+    foreach ($Server in $CheckAuthCertDistribution) {
+        $ServerName = ($Server -split "\.")[0]
+        Write-Host -ForegroundColor White  "  Server: " $ServerName
+        #Write-Host  "   Thumbprint: " $Thumbprint
+        if ($Server.Thumbprint -like $thumbprint) {
+            Write-Host  "   Thumbprint: "$Server.Thumbprint
+            Write-Host  "   Subject: "$Server.Subject
+            $ServerIdentity = $Server.Identity
+            $Script:tdCheckAuthCertDistribution = "   <div>Certificate with Thumbprint: $($Server.Thumbprint) Subject: $($Server.Subject) is present in Server $ServerIdentity</div>"
+            $Script:tdCheckAuthCertDistributionColor = "green"
+        }
+        if ($Server.Thumbprint -ne $thumbprint) {
+            Write-Host -ForegroundColor Red "  Auth Certificate seems Not to be present in $ServerName"
+            $Script:tdCheckAuthCertDistribution = "   Auth Certificate seems Not to be present in $ServerName"
+            $Script:tdCheckAuthCertDistributionColor = "Red"
+        }
+    }
+    CurrentCertificateThumbprintCheckHtml
+}
+function OAuthConnectivityCheck {
+    Write-Host -ForegroundColor Green " Test-OAuthConnectivity -Service EWS -TarGetUri https://outlook.office365.com/EWS/Exchange.asmx -Mailbox $Script:UserOnPrem"
+    PrintDynamicWidthLine
+    $Script:OAuthConnectivity = Test-OAuthConnectivity -Service EWS -TarGetUri https://outlook.office365.com/EWS/Exchange.asmx -Mailbox $Script:UserOnPrem
+    if ($Script:OAuthConnectivity.ResultType -eq 'Success' ) {
+        #$Script:OAuthConnectivity.ResultType

```

Comment:
Changed to:

![image](https://github.com/user-attachments/assets/94748b50-4225-4938-9731-f280ab7a7af3)

In case OAuth test is not successful we output the enitre error. Success case is handled latter in code.